modules: hostap: 11k/v/r roaming support

nxf58150 opened the pull request:
Added 11k/v/r roaming support.

nxf58150 committed:
modules: hostap: add 11k cmd support

Add 11k cmds support. User can issue 11k cmds to enable/disable 11k and
send Neighbor Report Request packet.

Signed-off-by: Hui Bai <hui.bai@nxp.com>

nxf58150 committed:
modules: hostap: add IEEE80211R support

Add 80211R support in hostap.
Add cmd wifi connect option '-R' to enable 80211R.

Signed-off-by: Hui Bai <hui.bai@nxp.com>

nxf58150 committed:
modules: hostap: add 11k/v/r roaming support

Added new ops and events in glue layer to support roaming.
Added new flag WIFI_NM_WPA_SUPPLICANT_ROAMING to control roaming
feature.

Signed-off-by: Hui Bai <hui.bai@nxp.com>

nxf58150 committed:
net: Increase net_mgmt task priority

Originally, the net_mgmt task priority is very low. Based on roaming
implementation, roaming is triggered in net_mgmt task. When running UDP
test while doing roaming, the net_mgmt task won't have much chance to
run and roaming can't be triggered.
Increase it to 3, which is same value of supplicant task.

Signed-off-by: Hui Bai <hui.bai@nxp.com>

nxf58150 committed:
manifest: Update hostap repo

Update hostap repo in west.yml

Signed-off-by: Hui Bai <hui.bai@nxp.com>

nxf58150 commented:
The 11kvr roaming RFC link: https://github.com/zephyrproject-rtos/zephyr/issues/76307

jukkar commented on subsys/net/l2/wifi/wifi_shell.c:
```diff
@@ -2909,6 +2978,10 @@ SHELL_STATIC_SUBCMD_SET_CREATE(wifi_commands,
 		cmd_wifi_btm_query,
 		2, 0),
 #endif
+	SHELL_CMD_ARG(11k_enable, NULL, "<0/1>\n",
```
I am not a big fan of just giving 0/1 here, it would be more logical if user could just say "enable" or "disable".
So `wifi 11k enable` or `wifi 11k disable` or similar.

nxf58150 replied:
Updated. The new command "wifi 11k [enable/disable]" can be used to enable or disable 11k, instead of giving paramter 0/1.

jukkar commented on subsys/net/l2/wifi/wifi_shell.c:
```diff
@@ -1021,6 +1021,75 @@ static int cmd_wifi_stats(const struct shell *sh, size_t argc, char *argv[])
+		}
+		(void)memcpy((void *)params.ssid, (const void *)argv[2],
+			     (size_t)strlen(argv[2]));
+
```
Remove extra empty line

nxf58150 replied:
Removed.

jukkar commented on subsys/net/l2/wifi/wifi_shell.c:
```diff
@@ -1021,6 +1021,75 @@ static int cmd_wifi_stats(const struct shell *sh, size_t argc, char *argv[])
+		return -ENOEXEC;
+	}
+
+	if (argc == 3)
+		PR("%s %s %s requested\n", argv[0], argv[1], argv[2]);
+	else
+		PR("%s requested\n", argv[0]);
```
`{}` are missing

nxf58150 replied:
Added` {}`

jukkar commented on modules/hostap/src/supp_api.c:
```diff
@@ -568,7 +568,8 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 
-			if (!wpa_cli_cmd_v("set_network %d key_mgmt SAE", resp.network_id)) {
+			if (!wpa_cli_cmd_v("set_network %d key_mgmt SAE%s", resp.network_id,
+			    params->ft_used ? " FT-SAE" : "")) {
```
Please align the params to `"set_network...`

nxf58150 replied:
Updated.

jukkar commented on modules/hostap/src/supp_api.c:
```diff
@@ -588,8 +589,8 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
-			if (!wpa_cli_cmd_v("set_network %d key_mgmt WPA-PSK",
-					   resp.network_id)) {
+			if (!wpa_cli_cmd_v("set_network %d key_mgmt WPA-PSK%s",
+			    resp.network_id, params->ft_used ? " FT-PSK" : "")) {
```
same here

nxf58150 replied:
Updated.

jukkar commented on modules/hostap/Kconfig:
```diff
@@ -249,6 +249,9 @@ if WIFI_NM_WPA_SUPPLICANT_CRYPTO_ENTERPRISE
 endif
 
+config WIFI_NM_WPA_SUPPLICANT_ROAMING
+	bool "Roaming support"
```
Could there be a help text also describing the supported roaming functionality

nxf58150 replied:
Added help text.

krish2718 replied:
needs an elaborate text to describe the `roaming`.

nxf58150 replied:
Updated.

jukkar commented on modules/hostap/src/supp_api.c:
```diff
@@ -75,6 +75,8 @@ static void supp_shell_connect_status(struct k_work *work);
 static K_WORK_DELAYABLE_DEFINE(wpa_supp_status_work,
 		supp_shell_connect_status);
 
+#define STR_CUR_TO_END(cur) (cur) = (&(cur)[0] + strlen((cur)))
```
This causes the length to be calculated multiple times when this is called, why not have a len parameter in this macro and calculate the value only once.

nxf58150 replied:
This macro define is not created in this PR. I just moved it to this location. This macro define was created and used for DPP feature.

jukkar replied:
As I said, this macro calculates string length multiple times without any apparent benefit. You do not need to copy some macro from other place just because it exists there.
I will add more comments below.

jukkar commented on modules/hostap/src/supp_api.c:
```diff
@@ -1139,6 +1149,72 @@ int supplicant_11k_neighbor_request(const struct device *dev, struct wifi_11k_pa
+int supplicant_spec_scan(const struct device *dev, struct wifi_scan_params *params)
+{
+	int i = 0;
+	char cmd[100] = {0};
+	char *pos = cmd;
+	char *end = pos + 100;
```
Could you have a define for the array length, or alternatively use sizeof. It is error prone to hard code the value as they can become mismatched if changed later.
So we could say here `char *end = pos + sizeof(cmd);`

nxf58150 replied:
Updated. Added new macro define for the array length.

jukkar commented on samples/net/wifi/boards/rd_rw612_bga.conf:
```diff
@@ -73,6 +73,8 @@ CONFIG_ZPERF_WORK_Q_THREAD_PRIORITY=3
 CONFIG_NET_SOCKETS_SERVICE_THREAD_PRIO=3
```
In commit `subsys: net: Increase net_mgmt task priority`, please remove the `subsys:` prefix as that is not needed to mention.

nxf58150 replied:
Removed `subsys:` in commit message.

nxf58150 commented on modules/hostap/src/supp_api.c:
```diff
@@ -1096,6 +1096,48 @@ int supplicant_pmksa_flush(const struct device *dev)
+int supplicant_11k_neighbor_request(const struct device *dev, struct wifi_11k_params *params)
+{
+	if (params && strlen(params->ssid)) {
+		if (strlen(params->ssid) > WIFI_SSID_MAX_LEN) {
```
Added a local variable for ssid length. Thanks!

nxf58150 commented on subsys/net/l2/wifi/wifi_shell.c:
```diff
@@ -1021,6 +1021,75 @@ static int cmd_wifi_stats(const struct shell *sh, size_t argc, char *argv[])
+
+	context.sh = sh;
+
+	if (argc != 2) {
```
Updated. The new command "wifi 11k" will be used to get 11k status (enabled or disabled) if no parameter was given.

krish2718 commented on include/zephyr/net/wifi_mgmt.h:
```diff
@@ -729,6 +743,16 @@ enum wifi_mgmt_op {
+	/** 11k enable/disable */
+	int enable_11k;
+	/** SSID */
+	uint8_t ssid[WIFI_SSID_MAX_LEN+1];
```
`WIFI_SSID_MAX_LEN + 1`

nxf58150 replied:
Updated.

krish2718 commented on modules/hostap/CMakeLists.txt:
```diff
@@ -219,6 +223,10 @@ zephyr_library_compile_definitions_ifdef(CONFIG_WIFI_NM_WPA_SUPPLICANT_AP
 	CONFIG_EAP_SERVER_IDENTITY
 )
 
+zephyr_library_sources_ifdef(CONFIG_IEEE80211R
+	${HOSTAP_SRC_BASE}/ap/wpa_auth_ft.c
+)
+
```
can be clubbed with above?

nxf58150 replied:
Updated.

krish2718 commented on include/zephyr/net/wifi_mgmt.h:
```diff
@@ -112,6 +112,12 @@ enum net_request_wifi_cmd {
+	/** Neighbor report complete */
+	NET_REQUEST_WIFI_CMD_NEIGHBOR_REP_COMPLETE,
+	/** Specific scan */
+	NET_REQUEST_WIFI_CMD_SPEC_SCAN,
```
can be confused with spectral scan (another feature in Wi-Fi). May be candidate scan?

nxf58150 replied:
Updated to candidate_scan.

jukkar commented on include/zephyr/net/wifi_mgmt.h:
```diff
@@ -729,6 +743,16 @@ enum wifi_mgmt_op {
+	/** 11k command operation */
+	enum wifi_mgmt_op oper;
+	/** 11k enable/disable */
+	int enable_11k;
```
For boolean type flags, please use `bool` type.

nxf58150 replied:
Updated to` bool` type.

jukkar commented on modules/hostap/src/supp_api.c:
```diff
@@ -1128,6 +1128,50 @@ int supplicant_pmksa_flush(const struct device *dev)
+{
+	int ssid_len = strlen(params->ssid);
+
+	if (params && ssid_len) {
```
There has been some effort in the past to write the non boolean type checks non boolean way. So what you have here works, it would be preferable to write this as
```
if (params != NULL && ssid_len > 0) {
```

nxf58150 replied:
Updated.

jukkar commented on modules/hostap/src/supp_api.c:
```diff
@@ -1128,6 +1128,50 @@ int supplicant_pmksa_flush(const struct device *dev)
+			return -1;
+		}
+
+		if (ssid_len) {
```
This check seems redundant, the ssid_len cannot be 0 at this point.

nxf58150 replied:
Removed this `ssid_len` check.

jukkar commented on subsys/net/l2/wifi/wifi_shell.c:
```diff
@@ -1021,6 +1021,84 @@ static int cmd_wifi_stats(const struct shell *sh, size_t argc, char *argv[])
+	}
+
+	if (params.oper == WIFI_MGMT_GET) {
+		PR("11k is %s\n", params.enable_11k == 0 ? "disabled" : "enabled");
```
If the `enable_11k` is boolean type, then you can say here `params.enable_11k ? "disabled" : "enabled");`

nxf58150 replied:
Updated.

jukkar commented on include/zephyr/net/wifi_mgmt.h:
```diff
@@ -486,6 +486,8 @@ struct wifi_connect_req_params {
 	/** Private key passwd length */
 	uint8_t key_passwd_length; /* Max 128 */
+	/** Fast BSS Transition used */
+	int ft_used;
```
This looks also boolean type, so please use `bool`

nxf58150 replied:
Updated to `bool`.

jukkar commented on modules/hostap/src/supp_api.c:
```diff
@@ -1173,6 +1183,73 @@ int supplicant_11k_neighbor_request(const struct device *dev, struct wifi_11k_pa
+	int freq = 0;
+
+	strcpy(pos, "freq=");
+	STR_CUR_TO_END(pos);
```
Here you could just say `pos += 5;` as we always copy the same string. There really is no need to do `strlen()`

nxf58150 replied:
Updated.

jukkar commented on modules/hostap/src/supp_api.c:
```diff
@@ -1173,6 +1183,73 @@ int supplicant_11k_neighbor_request(const struct device *dev, struct wifi_11k_pa
+	STR_CUR_TO_END(pos);
+	while (params->band_chan[i].channel) {
+		if (i > 0) {
+			snprintf(pos, end - pos, ",");
+			STR_CUR_TO_END(pos);
```
And this could be replaced by `pos += snprintf(pos, end - pos, ",");`

nxf58150 replied:
Updated.

jukkar commented on modules/hostap/src/supp_api.c:
```diff
@@ -1173,6 +1183,73 @@ int supplicant_11k_neighbor_request(const struct device *dev, struct wifi_11k_pa
+			STR_CUR_TO_END(pos);
+		}
+		freq = chan_to_freq(params->band_chan[i].channel);
+		snprintf(pos, end - pos, "%d", freq);
+		STR_CUR_TO_END(pos);
```
same here

nxf58150 replied:
Updated

jukkar commented on subsys/net/l2/wifi/wifi_mgmt.c:
```diff
@@ -386,6 +425,123 @@ void wifi_mgmt_raise_disconnect_result_event(struct net_if *iface, int status)
+	const struct device *dev = net_if_get_device(iface);
+	const struct wifi_mgmt_ops *const wifi_mgmt_api = get_wifi_api(iface);
+	struct wifi_scan_params params = {0};
+	int i = 0;
```
Unnecessary init here. You could replace this and the for loop by
```
for (int i = 0; i < roaming....) {
```

nxf58150 replied:
Updated.

jukkar commented on subsys/net/l2/wifi/wifi_mgmt.c:
```diff
@@ -18,6 +18,32 @@ LOG_MODULE_REGISTER(net_wifi_mgmt, CONFIG_NET_L2_WIFI_MGMT_LOG_LEVEL);
+};
+
+struct wifi_roaming_params {
+	uint8_t is_11r_used;
+	uint8_t is_11k_enabled;
```
These should probably be booleans

nxf58150 replied:
Updated to `bool`.

jukkar commented on subsys/net/l2/wifi/wifi_mgmt.c:
```diff
@@ -386,6 +425,123 @@ void wifi_mgmt_raise_disconnect_result_event(struct net_if *iface, int status)
+void wifi_mgmt_raise_neighbor_rep_recv_event(struct net_if *iface, char *inbuf, size_t buf_len)
+{
+	const uint8_t *buf = inbuf;
+	char event[32] = {0};
```
Please avoid having bare numbers because it is difficult to know what can be written to these. Have a define or something that describes better the value.

If the event is suppose to store ssid, then you need to have one extra byte for `\0`

nxf58150 replied:
Added new macro define for the max event string length.

jukkar commented on subsys/net/l2/wifi/wifi_mgmt.c:
```diff
@@ -386,6 +426,122 @@ void wifi_mgmt_raise_disconnect_result_event(struct net_if *iface, int status)
+{
+	const uint8_t *buf = inbuf;
+	char event[MAX_EVENT_STR_LEN] = {0};
+	char bssid[WIFI_SSID_MAX_LEN] = {0};
+	char bssid_info[WIFI_SSID_MAX_LEN]  = {0};
```
Is this correct? I mean that BSSID is typically the mac address so we should be able to have smaller buffer size here? Now we are allocating buffer of SSID length.

nxf58150 replied:
The bssid here is originally from supplicant API wpas_ctrl_neighbor_rep_cb(). After receiving and parsing Neighbor Report, wpa_supplicant will save all info of the report and the bssid will be saved as string here:
![image](https://github.com/user-attachments/assets/1bceef80-c020-4d54-ad66-7c019d9f5471)
Therefore, I reused SSID length for BSSID string.

jukkar commented on subsys/net/l2/wifi/wifi_mgmt.c:
```diff
@@ -386,6 +426,122 @@ void wifi_mgmt_raise_disconnect_result_event(struct net_if *iface, int status)
+	if (!buf || buf[0] == '\0') {
+		return;
+	}
+	if (sscanf(buf, "%s bssid=%s info=%s op_class=%d chan=%d phy_type=%d",
```
Missed this in my earlier review, but this is a bit error prone as we could overwrite the buffer because there is no length for the string variables.
We could try something like (not tested)
```
if (sscanf(buf, "%" STRINGIFY(MAX_EVENT_STR_LEN) "s bssid=%" STRINGIFY(WIFI_BSSID_MAX_LEN) "s info=% " STRINGIFY(WIFI_BSSID_MAX_LEN) "s op_class=%d chan=%d phy_type=%d",
```

nxf58150 replied:
As I explained before, the bssid is saved as string. If I use STRINGIFY(WIFI_BSSID_MAX_LEN), the bssid length will be incorrect.

jukkar replied:
I would like to avoid situation where we could overwrite bssid buffer if too long bssid is given. That is why I suggested putting length to the sscanf call.
What do you mean by `bssid length will be incorrect.`, please explain?

nxf58150 replied:
The BSSID is saved as string in the buffer with format "XX:XX:XX:XX:XX:XX". The string length is 17 without '\0'.
`if (sscanf(buf, "%" STRINGIFY(MAX_EVENT_STR_LEN) "s bssid=%" STRINGIFY(WIFI_BSSID_MAX_LEN) "s info=% " STRINGIFY(WIFI_BSSID_MAX_LEN) "s ...`
To my understanding, for the above code, it will be translated as below (suppose WIFI_BSSID_MAX_LEN is 6):
 `if (sscanf(buf, "%32s bssid=%6s info=%6s ...`
 For BSSID, it will only read 6 characters instead of 17, right? For event, it will read 32 characters while the actual length of event string might be less than 32. If it is true, this length check might cause errors. 
 So I think may be we can keep it for now. If I have better solutions, I can do code enhancement in the future.

jukkar replied:
Sure, that is correct. In that case we could have here `bssid=%" STRINGIFY(WIFI_SSID_MAX_LEN)`
My point is that we should tell sscanf call what is the max size of the buffer we are working with so that it is not possible to overwrite the buffer and cause stack corruption.

jukkar replied:
Ok, lets go with this solution at this point. Please return to this in near future.

nxf58150 replied:
Yes, of course. I will try your suggestion locally. Thanks!

jukkar replied:
@nxf58150 I created an enhancement request to fix the string length issue in sscanf, please address this asap so that we can get a fix for this in 4.0 release.

jukkar commented on subsys/net/l2/wifi/wifi_mgmt.c:
```diff
@@ -386,6 +426,122 @@ void wifi_mgmt_raise_disconnect_result_event(struct net_if *iface, int status)
+
+		for (i = 0; i < roaming_params.neighbor_rep.neighbor_cnt; i++) {
+			if (strncmp((const char *)roaming_params.neighbor_rep.neighbor_ap[i].bssid,
+				    bssid, 32) == 0) {
```
Magic constant used, shouldn't this be the `sizeof(bssid)`

nxf58150 replied:
Updated.

jukkar commented on subsys/net/l2/wifi/wifi_mgmt.c:
```diff
@@ -386,6 +426,122 @@ void wifi_mgmt_raise_disconnect_result_event(struct net_if *iface, int status)
+		if (!match && (roaming_params.neighbor_rep.neighbor_cnt < MAX_NEIGHBOR_AP_LIMIT)) {
+			strncpy((char *)roaming_params.neighbor_rep.neighbor_ap[idx].bssid,
+				bssid, sizeof(roaming_params.neighbor_rep.neighbor_ap[idx].bssid));
+			len = strlen(bssid);
```
A bit of defensive programming, we could do
```
len = strnlen(bssid, sizeof(bssid));
```
here.

jukkar replied:
Alternatively we could just do `memcpy()` which would copy also the final `\0` in which case we also need to make sure that there is space for if in the buffer.

nxf58150 replied:
Updated.

jukkar commented on subsys/net/l2/wifi/wifi_mgmt.c:
```diff
@@ -386,6 +426,122 @@ void wifi_mgmt_raise_disconnect_result_event(struct net_if *iface, int status)
+			strncpy((char *)roaming_params.neighbor_rep.neighbor_ap[idx].bssid_info,
+				(bssid_info),
+				sizeof(roaming_params.neighbor_rep.neighbor_ap->bssid_info));
+			len = strlen(bssid_info);
```
same here

nxf58150 replied:
Updated.

jukkar commented on subsys/net/l2/wifi/wifi_mgmt.c:
```diff
@@ -386,6 +426,122 @@ void wifi_mgmt_raise_disconnect_result_event(struct net_if *iface, int status)
+			strncpy((char *)roaming_params.neighbor_rep.neighbor_ap[idx].bssid,
+				bssid, sizeof(roaming_params.neighbor_rep.neighbor_ap[idx].bssid));
+			len = strlen(bssid);
+			roaming_params.neighbor_rep.neighbor_ap[idx].bssid[len] = (uint8_t)'\0';
```
BTW, is there space for `\0` in the buffer or are we overwriting the last character always?

nxf58150 replied:
There is space for `\0`.

jukkar reviewed (commented):
Found some possible issues with string handling.

nxf58150 commented:
@jukkar 
Hi Jukka, could you please help review the PR again. I did some changes based on your latest comments. Thanks!

jukkar commented on subsys/net/l2/wifi/wifi_shell.c:
```diff
@@ -3245,6 +3323,17 @@ SHELL_SUBCMD_ADD((wifi), wps_pin, &wifi_commands,
 		 "[pin] Only applicable for set.\n",
 		 cmd_wifi_wps_pin,
 		 1, 1);
+SHELL_SUBCMD_ADD((wifi), 11k, &wifi_commands,
+		 "Configure 11k or get 11k status.\n"
+		 "[enable/disable]\n",
+		 cmd_wifi_11k,
+		 1, 1);
+
+SHELL_SUBCMD_ADD((wifi), 11k_neighbor_request, &wifi_commands,
+		 "Send Neighbor Report Request frame.\n"
+		 "[ssid <ssid>]\n",
+		 cmd_wifi_11k_neighbor_request,
+		 1, 2);
```
Could you place these in alphabetical order in the command list (so at top), it is easier to find the commands that way.

nxf58150 replied:
Updated.

jukkar commented on subsys/net/l2/wifi/wifi_mgmt.c:
```diff
@@ -21,6 +21,33 @@ LOG_MODULE_REGISTER(net_wifi_mgmt, CONFIG_NET_L2_WIFI_MGMT_LOG_LEVEL);
+#define MAX_EVENT_STR_LEN 32U
+
+struct wifi_rrm_neighbor_ap_t {
+	char ssid[WIFI_SSID_MAX_LEN];
```
Please allocate space for `'\0'`, so this needs to have +1

nxf58150 replied:
Updated

jukkar commented on subsys/net/l2/wifi/wifi_mgmt.c:
```diff
@@ -403,6 +443,122 @@ void wifi_mgmt_raise_disconnect_result_event(struct net_if *iface, int status)
+			strncpy((char *)roaming_params.neighbor_rep.neighbor_ap[idx].bssid_info,
+				(bssid_info),
+				sizeof(roaming_params.neighbor_rep.neighbor_ap->bssid_info));
+			len = strnlen(bssid, sizeof(bssid));
```
Should this be `strnlen(bssid_info, sizeof(bssid_info));`?
This could lead to missing `'\0'`, we should take into account the terminating null.
```
len = strnlen(bssid_info, sizeof(bssid_info) - 1);
```
Similar issue in line 535

nxf58150 replied:
Updated.

nxf58150 commented:
@jukkar 
Hi Jukka, I have this build error of my roaming PR. Do you know the cause of the build error?
![image](https://github.com/user-attachments/assets/a573ffc2-3cae-4d0f-a93e-f2d2a1fa4afb)
It happened today and no such error previously.

jukkar commented:
> It happened today and no such error previously.

Yes, the reason is that we now compile test hostap properly (done by  #79973), I mentioned this yesterdays wifi meeting.
Your changes cause this issue to be seen (probably indirectly), you need to just find where the `aes_unwrap` is defined and add it to the compilation.

jukkar commented:
Hmm, the `aes_wrap()` is actually mentioned in the list at
https://github.com/zephyrproject-rtos/zephyr/blob/1689eaf6a4551165905979772d981ed6713acef5/modules/hostap/CMakeLists.txt#L247
I wonder what is going on here. I will check what is going on here.

jukkar commented:
Found the aes_unwrap issue. This will solve it
```
diff --git a/modules/hostap/CMakeLists.txt b/modules/hostap/CMakeLists.txt
index efddecaf250..d5f9c97baef 100644
--- a/modules/hostap/CMakeLists.txt
+++ b/modules/hostap/CMakeLists.txt
@@ -251,8 +251,10 @@ zephyr_library_sources_ifdef(CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_NONE
 
        # FIXME: why do we need these when crypto is not selected?
        ${HOSTAP_SRC_BASE}/crypto/aes-wrap.c
+       ${HOSTAP_SRC_BASE}/crypto/aes-unwrap.c
        ${HOSTAP_SRC_BASE}/crypto/aes-internal.c
        ${HOSTAP_SRC_BASE}/crypto/aes-internal-enc.c
+       ${HOSTAP_SRC_BASE}/crypto/aes-internal-dec.c
        ${HOSTAP_SRC_BASE}/crypto/aes-omac1.c
        ${HOSTAP_SRC_BASE}/crypto/md5.c
        ${HOSTAP_SRC_BASE}/crypto/md5-internal.c
```
Could you add a commit to your PR for this, the commit could say something like
```
hostap: More fixes to crypto=none compilation

The aes_unwrap and aes_decrypt functions were missing if
CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_NONE is enabled.
Add missing C files to the compilation to fix this.
```

nxf58150 commented:
Ok, got it. Thanks!

nxf58150 committed:
hostap: More fixes to crypto=none compilation

The aes_unwrap() and aes_decrypt() functions were missing
if CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_NONE is enabled.
Add missing C files to fix this.

Signed-off-by: Hui Bai <hui.bai@nxp.com>

jukkar commented:
@nxf58150 the hostap PR is now merged, please update the manifest

nxf58150 commented:
@krish2718 
Hi Chaitanya, I rebased the PR today but your previous approvement was gone. Could you please help approve it again? And if everything is ok, could you please help merge it today? Thanks!

MaochenWang1 commented:
Hi @jukkar @krish2718 @dleach02 @danieldegrasse  please review this PR
